### DISCUSSION TITLE
Integration of AcidBox and Aciduino

### DISCUSSION THREAD

**Discussion by espcam** [copych/AcidBox#5] [Q&A] [answered]
@copych I've seen your questions regarding Aciduino compilation on ESP32, I'm also digging to this direction, but I didn't have any success. Can you please share your experience with ESP32-S2? 
I think if we get 2 ESP32 boards, DAC, power subsystem, 2.42'' OLED screen, expandable control board, this combo can become full featured groovebox. 
I've managed to add display for ESP32, but had no luck compiling Aciduino for ESP32-S3. For 1306 SSD OLED display the code line will be 
U8G2_SSD1306_128X64_NONAME_F_SW_I2C u8g2(U8G2_R0, /* clock=*/ 16, /* data=*/ 15, /* reset=*/ U8X8_PIN_NONE);
All pins are assignable, so you need to define them directly for I2C. I picked 15 for data and 16 for clock. Hopefully this will be useful for anyone who want's to add bigger display.

**Comment by copych** [comment, marked as answer]
@espcam initially I thought of combining AcidBox with Aciduino v2, and I even made it work together on two devboards, but by that time I already used heavily modded AcidBanger, and it turned out that I like it better. Aciduino v2 seems to have an overloaded navigation, and I could hardly get to using it. It has probably a lot of features, but I wanted to make a toy to have some fun, maybe even without a display, just knobs, buttons and some leds, accu powered.
It seems to me now that it's possible to add some controls to AcidBox&AcidBanger duo to get what it was intended to be.
If you interested in running Aciduino v2 on ESP series, please, check issues there https://github.com/midilab/uClock/issues/18 , I have shared what I've found.
